sdk/test/trace/tracer_config_test.cc

```diff
@@ -0,0 +1,95 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#include "opentelemetry/sdk/trace/tracer_config.h"
+#include <gtest/gtest.h>
+#include <array>
+#include <string>
+#include <utility>
+#include <vector>
+#include "opentelemetry/common/attribute_value.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/nostd/unique_ptr.h"
+#include "opentelemetry/sdk/instrumentationscope/instrumentation_scope.h"
+#include "opentelemetry/sdk/instrumentationscope/scope_configurator.h"
+
+namespace trace_sdk             = opentelemetry::sdk::trace;
+namespace instrumentation_scope = opentelemetry::sdk::instrumentationscope;
+
+/** Tests to verify the basic behavior of trace_sdk::TracerConfig */
+
+TEST(TracerConfig, CheckDisabledWorksAsExpected)
+{
+  trace_sdk::TracerConfig disabled_config = trace_sdk::TracerConfig::Disabled();
+  ASSERT_FALSE(disabled_config.IsEnabled());
+}
+
+TEST(TracerConfig, CheckEnabledWorksAsExpected)
+{
+  trace_sdk::TracerConfig enabled_config = trace_sdk::TracerConfig::Enabled();
+  ASSERT_TRUE(enabled_config.IsEnabled());
+}
+
+TEST(TracerConfig, CheckDefaultConfigWorksAccToSpec)
+{
+  trace_sdk::TracerConfig default_config = trace_sdk::TracerConfig::Default();
+  ASSERT_TRUE(default_config.IsEnabled());
+}
+
+/** Tests to verify the behavior of trace_sdk::TracerConfig::DefaultConfigurator */
+
+static std::pair<opentelemetry::nostd::string_view, opentelemetry::common::AttributeValue> attr1 = {
+    "accept_single_attr", true};
+static std::pair<opentelemetry::nostd::string_view, opentelemetry::common::AttributeValue> attr2 = {
+    "accept_second_attr", "some other attr"};
+static std::pair<opentelemetry::nostd::string_view, opentelemetry::common::AttributeValue> attr3 = {
+    "accept_third_attr", 3};
+
+static instrumentation_scope::InstrumentationScope test_scope_1 =
+    *instrumentation_scope::InstrumentationScope::Create("test_scope_1");
```

Comment:
It is weird to invoke Create() and then use the object behind the unique ptr, but given how InstrumentationScope does not expose public constructors, this is the best the test can do.

Approved as is then.